Restructuring

marco262 opened the pull request:
At the moment, the only file that is "reviewable" is config.ini. Every other file is currently a WIP.
See TSSSF/config.ini for an example of what a proper config file should look like.
Any feedback on better config file format, or code fixed to config.ini would be great appreciated!

andreymal committed:
fix of trimming cards

andreymal committed:
trimming dirty fix

andreymal committed:
Merge branch 'develop' into fixes

marco262 committed:
Merge pull request #4 from andreymal/fixes

Fixes (for develop)

marco262 committed:
Bring github up-to-date with all Dropbox changes

marco262 committed:
Restructuring .gitignore for changes

marco262 committed:
Moving files to their new homes

marco262 committed:
Restructured the code

Moved all handling of paths to the CardGen script
Removed the need for the first line of PON files to identify the CardGen
script
Other fixes

pixel-prism committed:
Testing things

marco262 committed:
Undoing the test

marco262 committed:
First run at adding config file

Failed attempt. Need to refactor.

PictureElements committed:
Added BaBOC Symlinks

marco262 committed:
Moved CardGen to __init__.py

marco262 committed:
Split up config file and resources dict

Moved font loading to its own function
Added the option to not preload assets

marco262 committed:
Created first mockup of configs

Offered two config options

marco262 committed:
Trimmed some unnecessary prefixes

marco262 committed:
Added caching

Removed preloading in favor of caching whenever a resource is needed.
Changed "resources" to "resources_cache"

marco262 committed:
Revert "Added caching"

This reverts commit b67a78653ea30f259449abcc924116afcc5b2022.

marco262 committed:
Added caching

Removed preloading in favor of caching whenever a resource is needed.
Renamed "resources" to "resources_cache".

marco262 committed:
Added example BaBOC configs

marco262 committed:
Bug fix in getfont()

Fixed bug where the wrong option name was being passed to other
functions in a few places

marco262 committed:
Absolute path of fonts in loadfont()

Makes for more descriptive error messages

marco262 committed:
Config file change

Resource paths now need the game folder included
Font overrides need both the font file and size

marco262 committed:
improved config print functions

marco262 committed:
Added symbols, colors, frames, backs, and anchors

marco262 committed:
Made several functions private

Made "Card Defaults" a constant
Made several functions private

marco262 committed:
Added module docstring

marco262 committed:
Fixed bug when checking if anchor is a sequence

marco262 committed:
Added some better demo code to GameGen

marco262 committed:
WIP symlink script

coandco commented on config_helper.py:
```diff
@@ -0,0 +1,300 @@
+    Attempts to return the value for the given option, first from a section
+        matching card_name, then from a section matching card_type, then
+        from the Defaults section. If none of the sections (if they exist)
+        have that option, then None is returned.
```
It looks to me like the fallback functionality described here isn't implemented yet.  Is that correct?

marco262 replied:
This documentation is out of date and needs to be updated. get() only looks at a particular section, right now. Other functions will use find_files_in_config before calling get(), so they also know where in the resources cache to add the new data.

coandco commented on config_helper.py:
```diff
@@ -0,0 +1,300 @@
+    return font
+
+def _find_option_in_config(option, card_type=None, card_name=None):
+    if config.has_section(card_name) and config.has_option(card_name, option):
```
I still think it'd be a good idea to add prefixes for type and name so you don't end up with naming conflicts, i.e. "type.pony" or "name.Moe Fluttershy".  That way, if you want to have a card named "Pony" or "Goal" or something, there's no unexpected consequences.

marco262 replied:
Good idea! I'd forgotten about this suggestion before, but this will go on the to-do list for when I get home.

coandco commented on config_helper.py:
```diff
@@ -0,0 +1,300 @@
+    Checks for an image in the resources_cache dict. If it's not there,
+    checks the config file and attempts to load the image into the
+    resources_cache directory. 
+    '''
```
I'm guessing that this function will only be used for reused assets like card frames and symbols?  I'm not sure how much sense it makes to try using this for individual card art.

marco262 replied:
Correct. You've reminded me that I'll need to make a separate get function for retrieving card art, something that knows in what directory the card art is stored. Probably getart() or something.

pixel-prism commented on config_helper.py:
```diff
@@ -0,0 +1,300 @@
+    to be used with GameGen.py.
+config_helper supports two config files: One main file, and one optional file to be
+    found in the 'card set' folder. Any options in the optional config file will
+    overwrite identical options in the mail file. In this way, card sets can be
```
"mail" --> "main"

marco262 committed:
CR fixes

Added getart() function
Fixed docstring for get()
Added "Type." and "Card." prefixes to appropriate config sections

marco262 committed:
Merged GameGen from Restructuring_CR
